ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/StoreBackedCommandExecutor.scala

```diff
@@ -56,7 +57,13 @@ private[apiserver] final class StoreBackedCommandExecutor(
   ): Future[Either[ErrorCause, CommandExecutionResult]] = {
     val interpretationTimeNanos = new AtomicLong(0L)
     val start = System.nanoTime()
+    val coids = commands.commands.commands.toSeq.foldLeft(Set.empty[Value.ContractId]) {
+      case (acc, com.daml.lf.command.ApiCommand.Exercise(_, coid, _, argument)) =>
+        argument.collectCids(acc) + coid
+      case (acc, _) => acc
+    }
     for {
+      _ <- contractStore.prefetchContracts(coids.toSeq)
```

Comment:
I don't think that this is true. If you submit batches with 200 exercise commands, the overhead of running 200 individual DB queries compared to a single batch query is substantial. All the canton performance work so far was turning "single queries" into batch queries. So if this wouldn't be true here, then we wouldn't have seen any performance gain.

Reply:
I agree completely that query batching is important, beneficial and needed. 
I just don't understand how the practical usage would be: to hammer the command service, batching is not enough, you also have to do in parallel, otherwise the computation part of the engine is sequential, no? So you have to find a way to do batching plus parallelism. Also when you do parallel you immediately start to amortize away the DB roundtrip's overhead which would cool the effects of this improvement, right?